### PR TITLE
Ron azar/unit tests keren

### PR DESCRIPTION
**Pull request by RonAzar**

### CODE REVIEW

File: src/test/kotlin/com/hibob/academy/unitests/keren/ListManagerTest.kt [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+class ListManagerTest{
+    @Test
+    fun `Test adding a unique person`(){
+        assertEquals(true,ListManager().addPerson(Person("Ron Azar", 24)))
+    }
+
+    @Test
+    fun `Test adding a duplicate person and ensure it throws the expected exception`(){
+        assertThrows(IllegalArgumentException::class.java) {
```
Comment (kerensegev) [inline src/test/kotlin/com/hibob/academy/unitests/keren/ListManagerTest.kt:14]: You should better move these 2 first lines before the assert
            val managers = ListManager()
            managers.addPerson(Person("Ron Azar", 24))

This way you'll verify where the exception was thrown from

File: src/test/kotlin/com/hibob/academy/unitests/keren/ListManagerTest.kt [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        managers.addPerson(Person("Or Azar", 26))
+        assertEquals(2,managers.getManagersAmount())
+    }
+
+    //* Write tests for removePerson method:
+    @Test
+    fun `Test removing a person that exists in the list`(){
+        val managers = ListManager()
+        managers.addPerson(Person("Ron Azar", 24))
+        assertEquals(true,managers.removePerson(Person("Ron Azar",24)))
```
Comment (kerensegev) [inline src/test/kotlin/com/hibob/academy/unitests/keren/ListManagerTest.kt:34]: Need to make sure it was actually removed

File: src/main/kotlin/com/hibob/academy/unitests/keren/ListManager.kt [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    private val people: MutableList<Person> = mutableListOf()
+
+    fun addPerson(person: Person): Boolean {
+        if (people.any { it.name == person.name && it.age == person.age }) {
+            throw IllegalArgumentException("Duplicate person cannot be added.")
+        }
+        return people.add(person)
+    }
+
+    fun getManagersAmount(): Int{
```
Comment (kerensegev) [inline src/main/kotlin/com/hibob/academy/unitests/keren/ListManager.kt:47]: They are not managers, they are simple people :)
 ListManager means it manages a list of people

File: src/test/kotlin/com/hibob/academy/unitests/keren/ListManagerTest.kt [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        assertEquals(person3, managers.getPeopleSortedByAgeAndName()[0])
+        assertEquals(person2, managers.getPeopleSortedByAgeAndName()[2])
+    }
+
+
+    //calculateStatistics
+    @Test
+    fun `Test with an empty list for calculateStatistics`()
+    {
+        assertEquals(null, ListManager().calculateStatistics())
```
Comment (kerensegev) [inline src/test/kotlin/com/hibob/academy/unitests/keren/ListManagerTest.kt:105]: You can also use assertNull

**Review by kerensegev — APPROVED** [review]

File: src/main/kotlin/com/hibob/academy/unitests/keren/ListManager.kt (hunk 3 above)
Comment (RonAzar) [reply src/main/kotlin/com/hibob/academy/unitests/keren/ListManager.kt:47]: You're right!

File: src/test/kotlin/com/hibob/academy/unitests/keren/ListManagerTest.kt (hunk 2 above)
Comment (RonAzar) [reply src/test/kotlin/com/hibob/academy/unitests/keren/ListManagerTest.kt:34]: Improved

**Pull request merged**